api/client/src/main/java/org/projectnessie/client/http/NessieHttpClientBuilderImpl.java

```diff
@@ -269,21 +269,30 @@ public <API extends NessieApi> API build(Class<API> apiVersion) {
       builder.addTracing();
     }
 
+    NessieApiCompatibilityFilter nessieApiCompatibilityFilter = null;
     if (apiVersion.isAssignableFrom(HttpApiV1.class)) {
       if (enableApiCompatibilityCheck) {
-        builder.addRequestFilter(new NessieApiCompatibilityFilter(builder, 1));
+        nessieApiCompatibilityFilter = new NessieApiCompatibilityFilter(1);
+        builder.addRequestFilter(nessieApiCompatibilityFilter);
       }
       builder.setJsonView(Views.V1.class);
       HttpClient httpClient = HttpClients.buildClient(tracing, builder);
+      if (nessieApiCompatibilityFilter != null) {
+        nessieApiCompatibilityFilter.setHttpClient(httpClient);
```

Comment:
Could we instead share the `HttpAuthentication` between two `HttpClient` instances (the main one and the one used for API versions checks)?

Reply:
Another can of worms, then the `HttpAuthentication` instances need to track how often those are used.